README.md

```diff
@@ -2,11 +2,876 @@
+
+  @Autowired
+  private Tracer tracer;
+
+  private static HttpTextFormat.Setter<HttpRequest> setter =
+      new HttpTextFormat.Setter<HttpRequest>() {
+        @Override
+        public void set(HttpRequest carrier, String key, String value) {
+          carrier.getHeaders().set(key, value);
+        }
+      };
+
+
+  @Override
+  public ClientHttpResponse intercept(HttpRequest request, byte[] body,
+      ClientHttpRequestExecution execution) throws IOException {
+
+    Span currentSpan = tracer.getCurrentSpan();
+    currentSpan.setAttribute("client_http", "inject");
+    currentSpan.addEvent("Request sent to SecondService");
+
+    OpenTelemetry.getPropagators().getHttpTextFormat().inject(Context.current(), request, setter);
+
+    ClientHttpResponse response = execution.execute(request, body);
+
+    return response;
+  }
+}
+
+```
+
+```java
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+public class RestClientConfig {
+
+  @Autowired
+  RestTemplateHeaderModifierInterceptor restTemplateHeaderModifierInterceptor;
+
+  @Bean
+  public RestTemplate restTemplate() {
+    RestTemplate restTemplate = new RestTemplate();
+
+    List<ClientHttpRequestInterceptor> interceptors = restTemplate.getInterceptors();
+    if (interceptors.isEmpty()) {
+      interceptors = new ArrayList<>();
+    }
+    interceptors.add(restTemplateHeaderModifierInterceptor);
+    restTemplate.setInterceptors(interceptors);
+
+    return restTemplate;
+  }
+}
+```
+
+### Create a distributed trace 
+
+By default Spring Boot runs a Tomcat server on the port 8080. This tutorial assumes FirstService runs on the default port (8080) and SecondService runs on port 8081. This is because we hard coded the SecondService end point in FirstServiceController.SS_URL. To run SecondServiceApplication on port 8081 include `server.port=8081` in the resources/application.properties file. 
+
+Run both the FirstService and SecondService projects in terminal or using an IDE (ex. Eclipse). The end point for FirstService should be `http://localhost:8080/message` and `http://localhost:8081/time` for SecondService. Type both urls in a browser and ensure you receive a 200 response. 
+
+To visualize this trace add a trace exporter to one or both of your applications. Instructions on how to setup LogExporter and Jaeger can be seen [above](#tracer-configuration). 
+
+To create a distributed trace enter `http:\\localhost:8080/time` in a browser. This trace should include a span for FirstService and a span for SecondService.
+
+## Sample application with distributed tracing: otel-example 
```

Comment:
Is this a brand new top-level section? Shouldn't it be ### if you're still under the interceptors version?

Reply:
Yup, this is my food service example. It's still limping on. I might kill it for good if I can't get it past code review soon.

Reply:
My concern it that there are too many examples. I would personally completely drop the food service example for now.

When we get to a point where we have a fully out-of-the-box experience on par with Cloud Sleuth, then we'd put your demo code as a full instrumented example that people can experiment with. But let's not bind the starter project into the tutotorial. I believe Ray had made a similar comment about not wanting to see the starter project in what you submit to Github until we actually need a full sample instrumented.